forge/ee/routes/flowTemplates/index.js

```diff
@@ -0,0 +1,238 @@
+            query: { $ref: 'PaginationParams' },
+            response: {
+                200: {
+                    type: 'object',
+                    properties: {
+                        meta: { $ref: 'PaginationMeta' },
+                        count: { type: 'number' },
+                        templates: { $ref: 'FlowTemplateSummaryList' }
+                    }
+                },
+                '4xx': {
+                    $ref: 'APIError'
+                }
+            }
+        }
+    }, async (request, reply) => {
+        const paginationOptions = app.getPaginationOptions(request)
+        let filter = { active: true }
+        if (request.query.filter === 'all') {
+            filter = {}
+        } else if (request.query.filter === 'active') {
+            // Default behaviour
+            filter = { active: true }
+        } else if (request.query.filter === 'inactive') {
+            filter = { active: false }
+        }
+        const flowTemplates = await app.db.models.FlowTemplate.getAll(paginationOptions, filter)
+        flowTemplates.types = flowTemplates.templates.map(ft => app.db.views.FlowTemplate.flowTemplateSummary(ft))
+        reply.send(flowTemplates)
+    })
+
+    app.get('/:flowTemplateId', {
+        preHandler: app.needsPermission('flow-template:read'),
+        schema: {
+            summary: 'Get full details of a flow template',
+            tags: ['Flow Templates'],
+            params: {
+                type: 'object',
+                properties: {
+                    flowTemplateId: { type: 'string' }
+                }
+            },
+            response: {
+                200: {
+                    $ref: 'FlowTemplate'
+                },
+                '4xx': {
+                    $ref: 'APIError'
+                }
+            }
+        }
+    }, async (request, reply) => {
+        const flowTemplate = await app.db.models.FlowTemplate.byId(request.params.flowTemplateId)
+        if (!flowTemplate) {
+            reply.code(404).send({ code: 'not_found', error: 'Not Found' })
+            return
+        }
+        const response = app.db.views.FlowTemplate.flowTemplate(flowTemplate)
+        reply.send(response)
+    })
+
+    app.delete('/:flowTemplateId', {
+        preHandler: app.needsPermission('flow-template:delete'),
+        schema: {
+            summary: 'Delete a flow template - admin-only',
+            tags: ['Flow Templates'],
+            params: {
+                type: 'object',
+                properties: {
+                    flowTemplateId: { type: 'string' }
+                }
+            },
+            response: {
+                200: {
+                    $ref: 'APIStatus'
+                },
+                '4xx': {
+                    $ref: 'APIError'
+                }
+            }
+        }
+    }, async (request, reply) => {
+        const flowTemplate = await app.db.models.FlowTemplate.byId(request.params.flowTemplateId)
```

Comment:
Just checking, because this is an admin route, there's no need to validate who owns/created the flow right?

Reply:
Correct - at this stage the templates are owned by the platform, no finer grained permissions intended. Including a 'createdBy' field was more for info and possible future use.